lib/streamlit/runtime/state/common.py

```diff
@@ -215,7 +215,7 @@ def user_key_from_widget_id(widget_id: str) -> str | None:
     "None" as a key, but we can't avoid this kind of problem while storing the
     string representation of the no-user-key sentinel as part of the widget id.
     """
-    user_key = widget_id.split("-", maxsplit=2)[-1]
+    user_key: str | None = widget_id.split("-", maxsplit=2)[-1]
```

Comment:
this is an unrelated drive-by change since the linter complained about wrong typing